src/udisksdaemon.c

```diff
@@ -1329,6 +1328,44 @@ udisks_daemon_wait_for_object_sync (UDisksDaemon          *daemon,
   return ret;
 }
 
+UDisksObject *
+udisks_daemon_wait_for_object_sync (UDisksDaemon               *daemon,
+                                    UDisksDaemonWaitFuncObject  wait_func,
+                                    gpointer                    user_data,
+                                    GDestroyNotify              user_data_free_func,
+                                    guint                       timeout_seconds,
+                                    GError                      **error)
+{
+  UDisksObject *ret;
+
+  ret = wait_for_objects (daemon,
+                          (UDisksDaemonWaitFuncGeneric) wait_func,
+                          user_data,
+                          user_data_free_func,
+                          timeout_seconds,
+                          error);
+  return (UDisksObject *) ret;
```

Comment:
The type cast should be either dropped or moved above when the value is stored in the variable. Or just use ``return (UDisksObject *) wait_for_objects (...)``